templates/rust/common.tera

```diff
@@ -115,31 +115,86 @@ pub struct RegValueT<Reg: sealed::RegSpec> {
 }
 
 pub trait RegisterValue<T: RegSpec> {
+    /// Create a register value to be written to register from raw integer
+    ///
+    /// ```rust, ignore
+    /// // example with generic names
+    /// // needs: use test_pac::{timer, RegisterValue, TIMER}
+    /// let to_write = timer::BitfieldReg::new(0xdeadbeef);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// let to_write = to_write.boolw().set(true);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// ```
     #[must_use]
     fn new(data: T::DataType) -> Self;
-    /// Get raw register value
+
+    /// Unpack raw integer from value read from register
+    ///
+    /// ```rust,ignore
+    /// // example with generic names
+    /// // needs: use pac::{RegisterValue, TIMER}
+    /// let x = TIMER.bitfield_reg().read().get_raw();
+    /// ```
     #[must_use]
     fn get_raw(&self) -> T::DataType;
-    /// Return a copy with register value set to `value` and write mask fully set
+
+    /// Prepare value to written to register with an arbitrary value
+    ///
+    /// Use this function for setting a register to a custom value, independent
+    /// of bitfields, enumerations, etc. No checks are performed on the passed
+    /// value. The whole register is updated on write.
+    ///
+    /// ```rust,ignore
+    /// // example with generic names
+    /// // needs: use pac::{RegisterValue, TIMER}
+    /// TIMER.bitfield_reg().init(|r| r.set_raw(0xdeadbeef))
+    /// ```
     #[must_use]
     fn set_raw(self, value: T::DataType) -> Self;
 }
 
 impl<T: RegSpec> RegisterValue<T> for RegValueT<T> {
-/// Create a register value from raw value
+    /// Create a register value to be written to register from raw integer
+    ///
+    /// ```rust, ignore
+    /// // example with generic names
+    /// // needs: use pac::{timer, RegisterValue, TIMER}
+    /// let to_write = timer::BitfieldReg::new(0xdeadbeef);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// let to_write = to_write.boolw().set(true);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// ```
     #[inline(always)]
     fn new(data: T::DataType) -> RegValueT<T> {
         Self {
             data,
             mask: 0x0u8.into(),
         }
     }
-    /// Get raw register value
+
+    /// Unpack raw integer from value read from register
+    ///
+    /// ```rust,ignore
+    /// // example with generic names
+    /// // needs: use pac::{RegisterValue, TIMER}
+    /// let x = TIMER.bitfield_reg().read().get_raw();
+    /// ```
     #[inline(always)]
     fn get_raw(&self) -> T::DataType {
         self.data
     }
-    /// Return a copy with register value set to `value` and write mask fully set
+
+    /// Prepare value to written to register with an arbitrary value
```

Comment:
I would rephrase. "Prepare a register value that could be written to register ...."